py/mkrules.mk

```diff
@@ -134,7 +134,9 @@ $(OBJ): | $(HEADER_BUILD)/qstrdefs.generated.h $(HEADER_BUILD)/mpversion.h $(OBJ
 # See more information about this process in docs/develop/qstr.rst.
 $(HEADER_BUILD)/qstr.i.last: $(SRC_QSTR) $(QSTR_GLOBAL_DEPENDENCIES) | $(QSTR_GLOBAL_REQUIREMENTS)
 	$(ECHO) "GEN $@"
-	$(Q)$(PYTHON) $(PY_SRC)/makeqstrdefs.py pp $(CPP) output $(HEADER_BUILD)/qstr.i.last cflags $(QSTR_GEN_CFLAGS) cxxflags $(QSTR_GEN_CXXFLAGS) sources $^ dependencies $(QSTR_GLOBAL_DEPENDENCIES) changed_sources $?
+	$(Q)rm -f $(HEADER_BUILD)/qstr_sources.txt && \
+		for src in $^; do echo "$$src" >> $(HEADER_BUILD)/qstr_sources.txt; done && \
+		$(PYTHON) $(PY_SRC)/makeqstrdefs.py pp $(CPP) output $(HEADER_BUILD)/qstr.i.last cflags $(QSTR_GEN_CFLAGS) cxxflags $(QSTR_GEN_CXXFLAGS) sources @$(HEADER_BUILD)/qstr_sources.txt dependencies $(QSTR_GLOBAL_DEPENDENCIES) changed_sources $?
```

Comment:
Should we split this us into two recipes now that there is an intermediate file generated?